test/integ-new/main_new_test.go

```diff
@@ -0,0 +1,206 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This folder (test/integ-new) will be used to replace
+// folder test/integ after all new integ tests are implemented.
+// For now these tests will be ran by ci-new.yml.
+// If you are making changes to this file, please manully
+// run the ci-new workflow to make sure things works.
+
+// Main entry point for integration tests.
+package integration
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/abcxyz/jvs/pkg/cli"
+	"github.com/google/go-cmp/cmp"
+)
+
+var (
+	// Global integration test config.
+	cfg *config
+
+	// Keys we don't compare in validation result.
+	ignoreKeysMap map[string]struct{}
+)
+
+const (
+	serviceURLPrefix   = "https://"
+	jwkEndpointPostfix = ".well-known/jwks"
+	timestampFormat    = "2006-01-02 3:04PM UTC"
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(func() int {
+		ctx := context.Background()
+
+		ignoreKeysMap = map[string]struct{}{
+			"nbf": {},
+			"jti": {},
+		}
+
+		if strings.ToLower(os.Getenv("TEST_INTEGRATION")) != "true" {
+			log.Printf("skipping (not integration)")
+			// Not integration test. Exit.
+			return 0
+		}
+
+		// set up global test config.
+		c, err := newTestConfig(ctx)
+		if err != nil {
+			log.Printf("Failed to parse integration test config: %v", err)
+			return 2
+		}
+		cfg = c
+
+		return m.Run()
+	}())
+}
+
+func TestAPIAndPublicKeyService(t *testing.T) {
+	t.Parallel()
+	now := time.Now()
+	cases := []struct {
+		name                 string
+		ttl                  string
+		isBreakglass         bool
+		justification        string
+		wantValidationResult string
+	}{
+		{
+			name:          "none_breakglass",
+			justification: "issues/12345",
+			ttl:           "30m",
+			isBreakglass:  false,
+			wantValidationResult: fmt.Sprintf(`
+----- Justifications -----
+explanation    issues/12345
+
+----- Claims -----
+aud    [dev.abcxyz.jvs]
+exp    %s
+iat    %s
+iss    jvs.abcxyz.dev
+req    %s
+sub    %s
+`, now.Add(30*time.Minute).UTC().Format(timestampFormat), now.Format(timestampFormat), cfg.ServiceAccount, cfg.ServiceAccount),
+		},
+		{
+			name:          "breakglass",
+			justification: "issues/12345",
+			ttl:           "30m",
+			isBreakglass:  true,
+			wantValidationResult: fmt.Sprintf(`
+Warning! This is a breakglass token.
+
+----- Justifications -----
+breakglass    issues/12345
+
+----- Claims -----
+aud    [dev.abcxyz.jvs]
+exp    %s
+iat    %s
+iss    jvsctl
+sub
+`, now.Add(30*time.Minute).UTC().Format(timestampFormat), now.Format(timestampFormat)),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			var createCmd cli.TokenCreateCommand
+			_, stdout, _ := createCmd.Pipe()
+
+			server := fmt.Sprintf("%s---%s-%s:443", cfg.RevisionTagID, cfg.APIServiceName, cfg.ServiceURLPostfix)
```

Comment:
for this and the public key endpoint - I wonder if we can pass them as single env vars in the github workflow, if not, better to create them in https://github.com/abcxyz/jvs/pull/350/files#diff-134ca5370506bf2f23ad76f5bfce577db50a4137f3c139339f06cd5497603879R36

Reply:
Yes, this is doable, updated.